src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -2698,6 +2705,15 @@ internal bool CanDumpLibraryToXml(object obj)
 
         #region public methods
 
+        /// <summary>
+        /// Set UI culture based on preferences
+        /// </summary>
+        public static void SetUICulture(string locale)
+        {
+            Thread.CurrentThread.CurrentUICulture = new CultureInfo(locale == "Default" ? "en-US" : locale);
+            Thread.CurrentThread.CurrentCulture = new CultureInfo(locale == "Default" ? "en-US" : locale);
```

Comment:
These code were added in Dynamo 2.18 but removed in Dynamo 2.19. So this would mean adding these code back but conditionally (in-process integration)